pkg/apis/hive/v1/agent/platform.go

```diff
@@ -0,0 +1,68 @@
+package agent
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Platform stores cluster configuration for an agent based install where hosts are booted and registered, ready
+// to be assigned and used in a cluster.
+type Platform struct {
+	// Cluster name
+	Name string `json:"name"`
+	// Installation will start once cluster is in known state and approved
+	Approved                 bool   `json:"approved,omitempty"`
+	OpenshiftVersion         string `json:"openshiftVersion"`
+	BaseDnsDomain            string `json:"baseDnsDomain,omitempty"`
+	ClusterNetworkCidr       string `json:"clusterNetworkCidr,omitempty"`
+	ClusterNetworkHostPrefix int64  `json:"clusterNetworkHostPrefix,omitempty"`
+	ServiceNetworkCidr       string `json:"serviceNetworkCidr,omitempty"`
+	ApiVip                   string `json:"apiVip,omitempty"`
+	ApiVipDnsName            string `json:"apiVipDnsName,omitempty"`
+	IngressVip               string `json:"ingresVip,omitempty"`
+	MachineNetworkCidr       string `json:"machineNetworkCidr,omitempty"`
```

Comment:
We do not presently have any of these in the Hive API, we leave them to the install configs. Presumably now we'll need to add them, a question would be if they should be in an agent install platform, or higher level in the clusterdeployment itself, and only used for agent based install for now? For example, will some of these be useful for hypershift?

I'd be quite tempted here to model these to exactly match how the installer's InstallConfig does, move them higher up on clusterdeployment, and only let you specify them if platform = agentbased.

Properties need slight renames to match api conventions. IngressVIP, APIVIP, etc.

Reply:
After discussing with hive team, I've moved these to a BareMetal specific sub-struct. That struct is optional, but the fields within it are mandatory. This keeps the door open for non-metal specific agent based installs in the future.